packages/assets-controllers/src/TokensController.ts

```diff
@@ -648,9 +658,17 @@ export class TokensController extends BaseController<
 
       // We may be detecting tokens on a different chain/account pair than are currently configured.
       // Re-point `tokens` and `detectedTokens` to keep them referencing the current chain/account.
-      newTokens = newAllTokens?.[this.#chainId]?.[this.#selectedAddress] || [];
+      const currentInternalAccount = this.messagingSystem.call(
+        'AccountsController:getAccount',
+        this.#selectedAccountId,
+      );
+
+      // Previously selectedAddress could be an empty string. This is to preserve the behaviour
+      const currentAddress = currentInternalAccount?.address || '';
```

Comment:
```suggestion
      const selectedAddress = this.#getSelectedAddress();
```

Make sure to use `selectedAddresss` afterward (this would match the same naming as before)